src/baseclasses/chemical_energy/uvvismeasurementconcentration.py

```diff
@@ -212,12 +228,11 @@ def getConcentrationData(data_archive, logger, material, peak_value):
                         'The calibration is extrapolated based on the given material.'
                         'The computation of the concentration is based on the calibration linked in '
                         'the \'Concentration Detection\' reference section.')
+        concentration = calculate_concentration(calibration_entry['slope'],
```

Comment:
i dont get why you call it here and in L137. Couldnt you do it in such a way that you do it such that `getConcentrationData` only returns the reference and not the concentration? and then you calculate the concentration in the normalizer above?

by doing it like this you might forget that you need to change it in both functions when maintaining it? doe sit make sense?

Reply:
I thought I would call it here because the reference is already resolved and I have slope and so on. If you don't need the above with MProxy and it is resolved directly, you are right. I will change it.